pkg/client/client.go

```diff
@@ -33,6 +34,7 @@ type CortexClient struct {
 	id       string
 	endpoint *url.URL
 	client   http.Client
+	legacy   bool
```

Comment:
**suggestion:** store the actual path as a string here. It will reduce the amount of code you add in each function.

```suggestion
	apiPath string
```